### ISSUE TITLE
wisdom-asciidoc-maven-plugin has no support for asciidoct-diagram

### ISSUE THREAD

**Issue by Riduidel** [wisdom-framework/wisdom#566]
I would like to add plantuml diagrams in my asciidoc, bu it doesn't work with wisdom-asciidoc-maven-plugin.

Mimicking the asciidoctor-maven-plugin, I tried to add the asciidoctorj-diagram dependency, but it changed nothing.

So ...
Should I rewrite the wisdom-asciidoc-maven-plugin ? 
Should I create a wisdom-asciidotorc-maven-plugin ?
What is the best way to integrate diagrams in asciidoc generated by Wisdom ?

**Comment by cescoffier** [comment]
Did you try to update the "asciidoctor" version and then add the dependency ?

I never tried to integrate plantuml, but I agree it's pretty cool !

**Comment by Riduidel** [comment]
No, i didn't tried that, but I think there is another way to do it.
If I understand correctly, the wisdom-asciidoc-maven-plugin directly fires the asciidoc system when a modification is detected.

![What if I told you we can invoke maven-asciidoctor-plugin](https://cdn.meme.am/cache/instances/folder295/500x/74882295.jpg)

There is a library, called [Mojo Executor](http://timmoore.github.io/mojo-executor/) which allow one to call another plugin with configuration injected from Java code. using that library, it would be easy to have maven-asciidoc-plugin invoke the version of the asciidoctor-maven-plugin with the right set of options, and the right classpath.

What do you think ? Would it be a nice evolution of that plugin ?

**Comment by cescoffier** [comment]
MojoExecutor is nice but bring some classloading issue….

**Comment by Riduidel** [comment]
Well, in fact, I have two options :

1. use Mojo-Executor down the rabbit hole (dunno why you talk about Classloader issue : the other plugin has to be in MojoExecutor classpath, which means in plugin dependencies)
1. Directly extend the `AsciidoctorMojo`. I'm less fond of this approach, for ideological reasons (composition over inheritance, blablabla)

Which one do you prefer ?

**Comment by cescoffier** [comment]
I would go for the Mojo-Executor and see how it goes.

**Comment by Riduidel** [comment]
OK, I think I've tried to stretch my classloader-fu too much and got kicked in the balls.
As a consequence, I will downgrade my objective and just make sure our invocation of asciidoctor is clean enough to allow additional CLASSPATH to be used (and as a consequence allow asciidoctor-diagram to work, which was my initial goal).
